owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java

```diff
@@ -224,23 +225,30 @@ public void run() {
               .collect(Collectors.toList());
       Collections.shuffle(connections);
 
+      AtomicInteger leecherCount = new AtomicInteger(leechers.size());
```

Comment:
This runs in a single thread, so you don't need atomicity... The primitive integer would suffice